### PR TITLE
Add Qt5/Python3 builds to travis

### PR DESCRIPTION
**Pull request by m-kuhn**
This adds another build to the matrix which tests building with Qt5 and Python3.

At the moment is only a single test run (qgis_application). This already gives us useful hints when a change is introduced which is incompatible with Qt5.
## Server

The server test is disabled at the moment because it makes use of QFtp which isn't available in Qt5 out of the box. I am not sure how to proceed with this, but it's good to be aware of it ( @mhugent, @elpaso )
## Python

Now we can start exploring the automated migration and see how many python tests we get to pass this way and if there is more of the migration which needs to be (and can be) automatically updated.
## Failing tests

It's quite easy to turn the other tests on as well and check if you can fix one `ci/travis/linux/qt5/script.sh`

And here is the [cdash report of the status quo](http://dash.orfeo-toolbox.org/index.php?project=QGIS&filtercount=1&showfilters=1&field1=buildname/string&compare1=63&value1=d66452ed1b033f3c8ab46b096d3844dc8facd56a&date=2016-03-10)
## Pyspatialite

I wasn't able to compile pyspatialite so I turned it off. It may be responsible for one or the other failing test.
## QtWebkit

Some tests have been disabled because I have built Qt without webkit. We can hopefully fix that down the road.
## Technical background

Travis ships some libraries which are really, really old meanwhile and to use the container based builds we cannot use additional apt repositories without waiting for them to be whitelisted (and in the case of Qt5 this can take quite long because it does some setuid magic internally).

Long story short, I rebuilt all dependencies in the home folder and the dependency binaries get downloaded on every build. The build scripts and binaries are hosted at https://github.com/opengisch/osgeo4travis . Because it's a Dockerfile in there it should all be really easy to recreate all the dependencies.

### CODE REVIEW

**Comment by m-kuhn** [comment]
Maybe it would have been easier writing the formulas for homebrew, but I didn't know about it before...

No idea about the HTML report. The packed cmake from the cmake homepage is being used.

**Comment by m-kuhn** [comment]
@nyalldawson and thanks for every fixed image and test in advance!

**Comment by nyalldawson** [comment]
Hmm... can't work this out either. 

In other news, ctest/cdash is now top of my list for badly documented projects.

**Comment by mhugent** [comment]
>The server test is disabled at the moment because it makes use of QFtp which isn't available in Qt5 >out of the box. I am not sure how to proceed with this, but it's good to be aware of it ( @mhugent, >@elpaso )

Do you mean the QFtp used in QgsFtpTransaction? The server contains the (less known) functionality to load data from a remote server via http and ftp. It has been used in the university research project at that time. Don't know if anyone still uses it. For the switch to Qt5, it is best to just disable that part of code.

**Comment by m-kuhn** [comment]
Thanks @mhugent implemented in 3904722
